File: samples/scenarios/Scenario.Autofac/ViewModels/ShellViewModel.cs
using System;
using System.Threading;
using System.Threading.Tasks;
using Caliburn.Micro;

namespace Scenario.Autofac.ViewModels
{
    public class ShellViewModel : Screen
    {
        public ShellViewModel(ChildViewModel child)
        {
            Child = child;
        }

        protected override Task OnInitializedAsync(CancellationToken cancellationToken)
        {
            DisplayName = "Shell";
            return base.OnInitializedAsync(cancellationToken);
        }

        public ChildViewModel Child { get; }
    }
}

File: samples/scenarios/Scenario.Functional/ShellViewModel.fs
namespace Scenario.Functional.Core.ViewModels

open Caliburn.Micro
open System.Threading.Tasks

type ShellViewModel() = 
    inherit Screen()

    let mutable name = "Enter your name"

    override this.OnInitializedAsync (cancellationToken) =
        async {
            this.DisplayName <- "Shell"
        }
        |> Async.StartAsTask :> Task

    member this.Name
        with get() = name
        and set(value) =
            name <- value
            base.NotifyOfPropertyChange()

    member this.SayHello () =
        this.DisplayName <- "Hello " + this.Name
